fus.c

```diff
@@ -428,18 +429,74 @@ filelist_loadcb (Pool     *pool,
   return 1;
 }
 
+static gboolean
+download_to_path (const char   *url,
+                  const char   *path,
+                  SoupSession  *session,
+                  gboolean      force_download,
```

Comment:
instead of blindly re-downloading thing, please check hashsum referenced in repomd.xml, and if different -- redownload file.

Reply:
I'm just blindly re-downloading `repomd.xml`. For the other files, since the hashsum is in their name, I just check if the file referenced by repomd already exists.

Reply:
Fixed.